### DISCUSSION TITLE
Inserting into list from onChange event

### DISCUSSION THREAD

**Discussion by knownasilya** [edmundhung/conform#505] [Q&A] [answered]
```tsx
<SelectField
  labelProps={{ children: 'Topics', className: 'sr-only' }}
  selectProps={{
    defaultValue: '',
    placeholder: 'Select a topic',
    children: (
	<>
          <option value="">Select a topic</option>
          {topics.map(topic => (
	          <option key={topic.id} value={topic.id}>
		          {topic.name}
	          </option>
          ))}
	</>
    ),
    onChange: event => {
	const topicId = event.target.value
	const foundTopic = topics.find(t => t.id === topicId)
	
	list.insert(fields.topics.name, {
		defaultValue: foundTopic,
	})
    },
  }}
  />
```

The above insert doesn't seem to affect the ui at all, what could be the issue. I confirmed that onChange is called and the `foundTopic` is an object of shape `{ id, name }` which is the structure of `fields.topics`

**Comment by edmundhung** [comment, marked as answer]
It looks like you are still using `list` export from v0. The insert method will only returns all the button props required to execute an intent. You will need to dispatch it manually using the [requestIntent()](https://v0.conform.guide/intent-button#triggering-intent) helper

**Reply by knownasilya** [reply]
Glad v0 also has this helper, wasn't looking forward to the migration looks like it's pretty involved. Will have to dedicate some time for that. I wish the docs were a bit better with more examples and how to make custom inputs.

**Reply by jamalsoueidan** [reply]
I just converted v0 to v1, it took me 1 day and i have plenty of examples using mantine ui.

https://github.com/jamalsoueidan/booking-store/pull/29

**Comment by jamalsoueidan** [comment]
Which ui lib are you using?
